docs/devel/releasing.md

```diff
@@ -1,7 +1,156 @@
+1. `./build/mark-new-version.sh v${VER}.0` to mark the new release and get further
+   instructions. This creates a series of commits on the branch you're working
+   on (`release-${VER}`), including forking our documentation for the release,
+   the release version commit (which is then tagged), and the post-release
+   version commit.
+1. Follow the instructions given to you by that script. They are canon for the
+   remainder of the Git process. If you don't understand something in that
+   process, please ask!
+
+**TODO**: how to fix tags, etc., if you have to shift the release branchpoint.
+
+#### Building and Pushing Binaries
+
+In your git repo (you still have `${VER}` set from above right?):
+
+1. `git checkout upstream/master && build/build-official-release.sh v${VER}.0` (the `build-official-release.sh` script is version agnostic, so it's best to run it off `master` directly).
+1. Follow the instructions given to you by that script.
+1. At this point, you've done all the Git bits, you've got all the binary bits pushed, and you've got the template for the release started on GitHub.
+
+#### Writing Release Notes
+
+Finish the release notes using the [release notes tool](https://github.com/GoogleCloudPlatform/kubernetes/tree/master/contrib/release-notes) (**TODO**: Fill in details of how to do this easily.)
+
+### Building a New Patch Release (`vX.Y.Z` for `Z > 0`)
+
+#### Selecting Release Components
+
+For patch releases, the component selection and validation process is different
+(though see
+[https://github.com/GoogleCloudPlatform/kubernetes/issues/10676](#10676) for a
+way to unify them slightly more with `.0` releases). Patch releases are composed
+of cherry-picks. To create a candidate cherry-pick release and validate it, do
+the following:
+
+1. `export VER=x.y` (e.g. `0.20` for v0.20 - we don't care about the patch level
+   yet, we're just working on the release branch)
+1. `git checkout -b release-${VER} upstream/release-${VER}` (**NOTE:** This
+   assumes all release steps were followed for the `.0` release, which should
+   result in a `release-x.y` branch. If not, this step is recoverable, but stop
+   and ask someone.)
+1. Make sure you don't have any files you care about littering your repo (they
+   better be checked in or outside the repo, or the next step will delete them).
+1. `make clean && git reset --hard HEAD && git clean -xdf`
+1. `git cherry-pick -m 1 <merge commit of pull>` for each thing you need cherry
+   picked into the patch release. **NOTE**: Read carefully. You're selecting the
+   hash of the merge commit on `master`
+   (e.g. [974377b3](https://github.com/GoogleCloudPlatform/kubernetes/commit/974377b3064ac59b6e5694bfa568d67128026171)). This
+   keeps the PR #'s on the release branch for easy verification (at the mild
+   expense of squashing original authorship).
+1. Build and run e2e's: `go run hack/e2e.go -v -build -up -test -down |& tee /tmp/log`
```

Comment:
We run e2e before tagging and merging and such.  But as far as I can tell, there's no later "fetch the release tarball we pushed and do sanity checking on it" step?  E.g., chmods, unit, integration, e2e, $other_sanity_check?